rerun_cpp/src/rerun/time_column.hpp

```diff
@@ -0,0 +1,149 @@
+        /// Already sorted time points may perform better.
+        static TimeColumn from_sequence_points(
+            std::string timeline_name, Collection<int64_t> sequence_points,
+            SortingStatus sorting_status = SortingStatus::Unknown
+        ) {
+            return TimeColumn(
+                Timeline(std::move(timeline_name), TimeType::Sequence),
+                std::move(sequence_points),
+                sorting_status
+            );
+        }
+
+        /// Creates a sequence time column from an array of sequence points.
+        ///
+        /// \param timeline_name The name of the timeline this column belongs to.
+        /// \param timepoints_in_nanoseconds The time points in nanoseconds.
+        /// Make sure the sorting status is correctly specified.
+        /// \param sorting_status The sorting status of the time points.
+        /// Already sorted time points may perform better.
+        static TimeColumn from_times_nanoseconds(
+            std::string timeline_name, Collection<int64_t> timepoints_in_nanoseconds,
+            SortingStatus sorting_status = SortingStatus::Unknown
+        ) {
+            return TimeColumn(
+                Timeline(std::move(timeline_name), TimeType::Time),
+                std::move(timepoints_in_nanoseconds),
+                sorting_status
+            );
+        }
+
+        /// Creates a sequence time column from an array of arbitrary std::chrono durations.
+        ///
+        /// \param timeline_name The name of the timeline this column belongs to.
+        /// \param chrono_timepoints The time points as chrono durations.
+        /// Make sure the sorting status is correctly specified.
+        /// \param sorting_status The sorting status of the time points.
+        /// Already sorted time points may perform better.
+        template <typename TRep, typename TPeriod>
+        static TimeColumn from_times(
+            std::string timeline_name,
+            const Collection<std::chrono::duration<TRep, TPeriod>>& chrono_timepoints,
+            SortingStatus sorting_status = SortingStatus::Unknown
+        ) {
+            std::vector<int64_t> timepoints(chrono_timepoints.size());
+            for (size_t i = 0; i < chrono_timepoints.size(); i++) {
+                timepoints[i] =
+                    std::chrono::duration_cast<std::chrono::nanoseconds>(chrono_timepoints[i])
+                        .count();
+            }
+            return TimeColumn(
+                Timeline(std::move(timeline_name), TimeType::Time),
+                std::move(timepoints),
+                sorting_status
+            );
+        }
+
+        /// Creates a sequence time column from a range of sequence points.
+        ///
+        /// \param timeline_name The name of the timeline this column belongs to.
+        /// \param min The minimum sequence point, must be less than `max`.
+        /// \param max The maximum sequence point, must be greater than `min`.
```

Comment:
You should also clarify that `max` is exclusive.

You could use the words `begin` and `end` to match what C++ uses for iterators.

Reply:
if it's confusing we could also drop this method 🤷 
We don't have it on Python either